### PR TITLE
feature/support transit encoded inputs

### LINKED ISSUE #42: Not showing EDN from stdin
`pbpaste | floki` gets floki complaining about being unable to parse edn/json. OTOH `pbpaste > q; floki q` shows EDN no problems.

### LINKED ISSUE #46: Support transit encoded EDN
EDN seems to be rare as a wire format - might be wrong, but seems transit is more common as an encoding for sending EDN over the wire. 

Floki doesn't currently understand transit though, which means it can't directly consume the output of APIs that speak transit.

### PR DESCRIPTION
**Pull request by dazld**
Fixes #46 

Think it probably fixes #42 too, as looked like you were assuming there was a carriage return always...?  see what you think.

For reference: [MDN: String.prototype.trim](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim)

### CODE REVIEW

File: src/floki/core.cljs [hunk 1]
```diff
@@ -61,7 +68,7 @@
 
 (defn stdin-handler
   [buf]
-  (rf/dispatch [:input/set (convert buf)]))
+  (rf/dispatch [:input/set (-> buf convert)]))
```
Comment (denisidoro) [inline src/floki/core.cljs]: Why?

File: src/common/stdin.cljs [hunk 2]
```diff
@@ -16,5 +16,5 @@
   (.on stdin "end"
        (fn []
          ; (print "END EVENT!!!!")
-         (swap! stdinput #(->> % count dec (subs % 0)))
+         (swap! stdinput #(.trim %))
```
Comment (denisidoro) [inline src/common/stdin.cljs]: Cool!

Comment (denisidoro) [reply src/common/stdin.cljs]: Why not str/trim?

File: src/floki/core.cljs [hunk 3]
```diff
@@ -31,22 +32,28 @@
 
 (defn convert
   [x]
-  (let [res (try
+  (let [tr (transit/reader :json)
```
Comment (denisidoro) [inline src/floki/core.cljs]: Im not proud of this try/catch thing :(
Any idea on how to improve this?

File: src/common/stdin.cljs (hunk 2 above)
Comment (dazld) [reply src/common/stdin.cljs]: have to check what that does - this was first thing that came to mind. most likely str/trim is just wrapping the method from the prototype, so ends up being the same. I'll look!

File: src/floki/core.cljs (hunk 3 above)
Comment (dazld) [reply src/floki/core.cljs]: I mean.. it gets the job done :) Promises may help flatten it, and be a bit kinder to the event loop if there is more going on concurrently. Guess in this case would have to consider what other requirements there are. Reduce would work too!

File: src/floki/core.cljs (hunk 1 above)
Comment (dazld) [reply src/floki/core.cljs]: no particular reason - it's the same as the other handler like this, which is appealing. was playing with more steps here and it got left behind, I think.

File: src/common/stdin.cljs (hunk 2 above)
Comment (dazld) [reply src/common/stdin.cljs]: str/trim does some normalisation for browser inconsistencies and some obscure google thing about being a trusted site - without digging into what that means for CLJS, it's probably better to use the provided library function, so have updated to use that instead.

**Comment by dazld** [comment]
I'm out for a couple of days, but if this looks good as is, feel free to go ahead and merge! Thanks for the reviews :)

**Comment by denisidoro** [comment]
Sure! I'll try to merge it this week

**Pull request merged**